### ISSUE TITLE
Target output size mismatch

### ISSUE THREAD

**Issue by jaideep11061982** [fastai/fastai#1735]
**Describe the bug**

 input and target shapes do not match: input [48 x 5004], target [48] 
My network output is of output channel 5004 depending on no of classes I have.My netowork has to multiclass classfication not binary so  I was expecting category list class to do single class hot encoding  like the way it happens for Multicategory list for each label in the target. 

I suspect this is a bug as one may not have just binary classes ,there can be multiclasses as well. 

**Provide your installation details**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by sgugger** [comment]
In single classification, the labels aren't one-hot encoded since because the loss function of pytorch (`F.cross_entropy` and `F.nll_loss`) expect it that way.
